### DISCUSSION TITLE
Overlay: shop

### DISCUSSION THREAD

**Discussion by strct** [streetcomplete/StreetComplete#4231] [Ideas]
A shop overlay could help completing the shop coverage in osm. Currently there is no way to enter new shops in StreetComplete, but with an shop overlay that might change?

Opening a discussion first, because this is in the new quest guidelines:
> Only existing elements can be extended, no elements can be added or removed.[[1]](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)

Can this change with the overlay functionality and should I open an issue for this?

**Comment by Jean-BaptisteC** [comment]
Overlay already exists for shop, when you click on quests about shop, you see all shops around you.
Idem for bike parkings and cameras.

**Reply by legofahrrad** [reply]
No, overlays are something different. Overlays are a new feature since v45.

**Comment by matkoniecz** [comment]
> new quest guidelines

note that overlays and quests are distinct - and <s>[there are no overlay guidelines](https://github.com/streetcomplete/StreetComplete/blob/master/OVERLAY_GUIDELINES.md)</s> (though nowadays quests can remove nodes...)

EDITED: untrue claim stricken, linked https://github.com/streetcomplete/StreetComplete/blob/master/OVERLAY_GUIDELINES.md

**Reply by legofahrrad** [reply]
> there are no overlay guidelines

Yes there are overlay guidelines! See https://github.com/streetcomplete/StreetComplete/blob/master/OVERLAY_GUIDELINES.md

**Reply by matkoniecz** [reply]
...that I was aware of....

Thanks!

**Comment by peternewman** [comment]
> > Only existing elements can be extended, no elements can be added or removed.[[1]](https://github.com/streetcomplete/StreetComplete/blob/master/QUEST_GUIDELINES.md)
> 
> Can this change with the overlay functionality and should I open an issue for this?

The current two overlays don't implement that functionality. In theory you could press and hold like creating a note and add a different type of element as a node, but whether that's particularly efficient I don't know.

> A shop overlay could help completing the shop coverage in osm.

I think what it could do though, is help keep shop coverage up to date. Currently we can only mark a shop as gone/changed if there's an open quest, which generally means diet, wheelchair access, or card for non-chain stores. But see for example https://www.bbc.co.uk/news/business-60418435 and there is a constant churn regardless of shop type.

An overlay could let you select ANY shop on OSM and change it's type (like the current dialogue does). It could highlight vacant, chain/branded and standalone shops in different colours for a quick reference and perhaps permanently overlay the names in that mode.

**Comment by mnalis** [comment]
People interested in quickly adding POIs should also take a look at new [EveryDoor](https://wiki.openstreetmap.org/wiki/Every_Door) app; I find it much easier and faster to use than [Vespucci](https://wiki.openstreetmap.org/wiki/Vespucci) (which I still use for anything more complex or way / area related). It also allows for editing/updating many details of the POI at once (alleviating https://github.com/streetcomplete/StreetComplete/issues/124) and easy adding of multiple entrances / addresses, as well as offering multiple satellite maps overlays (https://github.com/streetcomplete/StreetComplete/issues/56)

There is quick (< 5 min!) and informative video tutorial by Ilya Zverev on that wiki page too demonstrating all the features of the app.
